FiniteStateMachine/src/demo/Car.java

```diff
@@ -35,12 +35,8 @@ public void reverse() {
         if (gearbox.getState() == Gear.REVERSE) {
             return;
         }
-        if (gearbox.nextStates().contains(Gear.REVERSE)) {
```

Comment:
`moveTo` already has logic to check the next state is valid and throw if it isn't, no need to duplicate that logic.